src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -148,6 +143,29 @@ private function shouldRetry(\Throwable $e, Envelope $envelope, RetryStrategyInt
         return $retryStrategy->isRetryable($envelope, $e);
     }
 
+    private function getWaitingTime(Envelope $envelope, \Throwable $throwable, RetryStrategyInterface $retryStrategy): int
+    {
+        $delay = null;
+        if ($throwable instanceof RecoverableExceptionInterface && method_exists($throwable, 'getRetryDelay')) {
+            $delay = $throwable->getRetryDelay();
+        }
+
+        if ($throwable instanceof HandlerFailedException) {
```

Comment:
Should we rather check for `WrappedExceptionsInterface`?

Reply:
I just used the same as in `\Symfony\Component\Messenger\EventListener\SendFailedMessageForRetryListener::shouldRetry`